.github/workflows/lint.yml

```diff
@@ -0,0 +1,27 @@
+name: lint
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+jobs:
+  golangci-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v4
+        with:
+          go-version-file: 'go.mod'
+
+      # This step sets up the variable steps.golangci-lint-version.outputs.v
+      # to contain the version of golangci-lint (e.g. v1.54.2).
+      # The version is taken from go.mod.
+      - name: Golangci-lint version
+        id: golangci-lint-version
+        run: |
+          GOLANGCI_LINT_VERSION=$( go list -m -f '{{.Version}}' github.com/golangci/golangci-lint )
+          echo "v=$GOLANGCI_LINT_VERSION" >> "$GITHUB_OUTPUT"
+      - name: golangci-lint
```

Comment:
Golangci-lint also requires a `.golangci.yml` file, see for instance what we have in [conduit-commons](https://github.com/ConduitIO/conduit-commons/blob/main/.golangci.yml). We'll also require a [`.golangci.goheader.template`](https://github.com/ConduitIO/conduit-commons/blob/main/.golangci.goheader.template) for the goheader linter.